server/lib.ts

```
@@ -0,0 +1,57 @@
+import { Prisma } from "@prisma/client";
+import { prisma } from "../lib/db";
+import { exit } from "process";
+import { createServer as createHttpsServer } from "node:https";
+import {
+  createServer as createHttpServer,
+  Server as HttpServer,
+} from "node:http";
+import { readFileSync } from "fs";
+import { RequestHandler } from "next/dist/server/next";
+
+export async function checkDatabaseConnection() {
+  return new Promise<void>(async (resolve, reject) => {
+    let connectionAttempts = 1;
+
+    while (connectionAttempts <= 3) {
+      try {
+        await prisma.$executeRaw(Prisma.sql`SELECT 1;`);
+        return resolve();
+      } catch (e) {
+        console.error(
+          `Database connection attempt ${connectionAttempts} failed, retrying...`,
+        );
+        await sleep(5000);
```

Should we bail immediately and not do the whole retry loop / is this delaying the inevitable?

---

My reasoning was just to give it a bit of time to get started so it wouldn't just fail instantly and keep getting redeployed if it had an issue on the first connection

---

Possibly? But in my head that'd only be the case if the *database server* is the one that's taking a while to start, which it should almost never be in production (the Prisma connection should block if it isn't ready yet), so we'd want to fail fast if something is up with the DB.

---

It's within 15 seconds, it's not going to cause a massive issue I don't think